### ISSUE TITLE
[Core] [Rouge] Rouges do not produce the expected number of auto attacks

### ISSUE THREAD

**Issue by Tamaxofalgalon** [dalaranwow/dalaran-wow#5834]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

Currently rouges are not doing around 5% of their auto attacks resulting in high impact in their dps as poisons and energy regeneration are severly gimped!

**Expected behaviour**: Tell us what should happen instead

Rouges should produce the expected number of auto attacks in a given time dictated by their main-hand and off-hand attack speed!

**Steps to reproduce the problem**:

1. Calculate the expected number of attacks in x seconds:
**(seconds / MH attack speed) + (seconds / OH attack speed)**

2. Attack a training dummy without any changes to your attack speed (haste buffs etc)

Stats while testing: expertise capped (as only 23 is needed here)
1.8 dagger in each hand - 1.47 attack speed after talents and haste
https://imgur.com/0bOreqe

3. See that you are doing only around 95% of the expected attacks

**Include proofs for this behaviour**
Screenshot of my second test (480 seconds of pure AA):
https://imgur.com/a/orcoVEe

As you can see the number of performed auto attacks is **624**
The expected value is (480 / 1.47) + (480 / 1.47) = **~653**
Difference = 29 = **~5%**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by ih8mondays** [comment]
If you want to improve your report you could do the same tests without talents, gear (starter daggers), offhand only, mainhand only etc. and compare your results to narrow down exactly where the issue lies.

**Comment by Tamaxofalgalon** [comment]
Follow up:

**All melee classes** (tested Paladin, Warrior, Death Knight) seem to suffer a comparable percentage of "non-existent" auto attacks which usually lies in the 2.5% to 5% range.
Tests done by me and guild mates indicate that a lower swing timer increases the fraction of attacks that are simply not made.

My theory is a **static delay** between swing timers. That would explain the increasing fraction of attacks lost with a lower swing timer.

Let us assume a theoretical "break" of 1 whole second in between swing timers and calculate the fraction of swing timers lost.

Swing timer of 3.0 → 1sec / 3.0 sec = **0.33**
For every attack you lose **one third** of an attack.

Swing timer of 1.0 → 1sec / 1.0 sec = **1**
For every attack you lose **one whole attack**.

Of course this is only an example but it should work the same way for a more realistic delay of 50 ms.

As rouges usually have the lowest swing timer of any melee class and rely on AAs for poisons, talent procs, energy regeneration and much more they suffer the greatest impact!

In my eyes this a game breaking issue that should be addressed immediately!
@Rushor @Lothloryen

Table of my performed tests:
https://docs.google.com/spreadsheets/d/1tep7WZfh0wqqnieQdTPig6aAQ4lKSENVkLWsLyCntJU/edit?usp=sharing

**Comment by mercred** [comment]
Duplicate of https://github.com/dalaranwow/dalaran-wow/issues/5677 and there is already a one week old comment of possible swing timer issue.  It is recommended to use search by keywords to save your own time and efforts. If you could perform more tests of static delay between swings theory that would be helpful.

**Comment by Tamaxofalgalon** [comment]
> 
> 
> Duplicate of #5677 and there is already a one week old comment of possible swing timer issue. It is recommended to use search by keywords to save your own time and efforts. If you could perform more tests of static delay between swings theory that would be helpful.

I disagree!
While being respectable for the sheer lenght and quality of testing done #5677 implies the issue stems form dualwielding when in fact all AAs (and classes) are affected.
